src/providers/oracle/qgsoraclesourceselect.h

```diff
@@ -134,7 +134,7 @@ class QgsOracleSourceSelect : public QDialog, private Ui::QgsDbSourceSelectBase
     //!Sets a new regular expression to the model
     void setSearchExpression( const QString& regexp );
 
-    void on_buttonBox_helpRequested() { QgsContextHelp::run( metaObject()->className() ); }
+    void on_buttonBox_helpRequested() { QgsHelp::openHelp( QStringLiteral( "working_with_vector/supported_data.html#id2" ) ); }
```

Comment:
To be honest, I still do not understand the way things are supposed to be implemented (sorry!). Will build this branch tonight :)
About the link, yes we should use more verbose link, because, if a new section is inserted in this chapter, the #id2 won't point to the same section